coffee-chats/src/main/java/com/google/step/coffee/servlets/ChatRequestServlet.java

```diff
@@ -0,0 +1,35 @@
+package com.google.step.coffee.servlets;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Date;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/chat-request")
+public class ChatRequestServlet extends HttpServlet {
+
+  @Override
+  protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+      throws ServletException, IOException {
+
+    String[] tags = req.getParameter("tags").split(",");
+    String datesString = req.getParameter("dates");
+    int numPeople = Integer.parseInt(req.getParameter("numPeople"));
+    int duration = Integer.parseInt(req.getParameter("duration"));
+    boolean randomMatch = Boolean.parseBoolean(req.getParameter("randomMatch"));
+    boolean pastMatched = Boolean.parseBoolean(req.getParameter("pastMatched"));
+
+    if (datesString.equals("")) {
```

Comment:
This will cause NullPointerException if certain parameters are not specified

Reply:
Done. I have added some wrapper functions to handle getting parameters.